utils/checks.stanza

```diff
@@ -381,34 +380,34 @@ defn cmos-and-ttl-checks (cmos-ttl-outputs:Tuple<DigitalPin>,
 ;
 defn open-collector-checks (open-collectors:Tuple<DigitalPin>,
                             pin-categories:DigitalPinCategories) :
-  val [pull-up-resistors, rail-voltages] = collect-pull-up-resistors-and-their-rail-voltages(open-collectors, pin-categories)
+  val [pull-up-resistors, net-voltages] = collect-pull-up-resistors-and-their-net-voltages(open-collectors, pin-categories)
   check-a-pull-resistor-exists(open-collectors, pull-up-resistors)
```

Comment:
The previous `propagate-rail-voltage` was finding supply powers and then propagating them on with rail-voltage properties. So replacing by `net-voltages` is incorrect. Every connected component pin is supposed to have a net voltage now so every resistor is detected as pull-up resistor and the check on pull-up resistor never fails

Reply:
If the user sets a voltage on the resistor pin that does not satisfy the vih of the input (after calculation), then the error will be triggered. We do calculate the current through the pull-up resistor and should use that current * resistance subtracted from the net-voltage to determine the input voltage.